lib/call-cabal-project-to-nix.nix

```diff
@@ -179,45 +180,36 @@ let
 
     # run `plan-to-nix` in $out.  This should produce files right there with the
     # proper relative paths.
-    (cd $out && plan-to-nix --plan-json $tmp/dist-newstyle/cache/plan.json -o .)
+    (cd $out && plan-to-nix --full --plan-json $tmp/dist-newstyle/cache/plan.json -o .)
```

Comment:
What's the `--full` option? Is it related to source filtering?

Reply:
Yes, it adds details to the Cabal2Nix output that we need for filtering [here](https://github.com/input-output-hk/nix-tools/blob/a9f227d9238102fb424157156712ef9b64413096/lib/Cabal2Nix.hs#L267-L273) and [here](https://github.com/input-output-hk/nix-tools/blob/a9f227d9238102fb424157156712ef9b64413096/lib/Cabal2Nix.hs#L301-L310)